### DISCUSSION TITLE
When TeferenceConfig.setApplication is Deprecated

### DISCUSSION THREAD

**Discussion by alpha2wang** [apache/dubbo#14376] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

I noticed that this method was first introduced from dubbo2.7.5 and annotated as deprecated. May I ask why so many versions have passed and not been discarded now? If discarded later, what is the alternative solution
```
@Deprecated
public void setApplication(ApplicationConfig application) {
    this.application = application;
    if (application != null) {
        this.getConfigManager().setApplication(application);
    }
}
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by aofall** [comment]
Use `org.apache.dubbo.config.AbstractConfig#setScopeModel` instead

**Comment by AlbumenJ** [comment]
Use `DubboBootstrap` to manage config instead
